File: Q2-Validate Sequence/Answer2.ts
export function isValidSubsequence(array: number[], sequence: number[]) {
    let seqIndex = 0
    const SEQUENCE_LENGHT = sequence.length
    
    for (let index = 0; index < array.length; index++) {
      if(SEQUENCE_LENGHT === seqIndex) return true;
  
      if(array[index] === sequence[seqIndex]){
        seqIndex+=1
      }
    }
    
    return SEQUENCE_LENGHT === seqIndex;
  }
  

File: Q2-Validate Sequence/Answer2.py
def isValidSubsequence(array, sequence):   
    indexSeq = 0
    seqLenght = len(sequence)
    for number in array:
        if(indexSeq == seqLenght):
            return True
            
        if(number == sequence[indexSeq]):
            indexSeq+=1


    return indexSeq == seqLenght
